rbc/src/main/java/com/library/rbc/service/BookService.java

```diff
@@ -94,7 +94,8 @@ private List<BookStatusDto> convertStringsToBookStatusesDto(List<String> bookSta
     }).toList();
   }
 
-  public String uploadImage(MultipartFile image) {
+  public String uploadImage(MultipartFile image, String id) {
+    BookDto bookDto = getBook(id);
```

Comment:
You can use the book repository for fetching the book in this context, then you don't need the mapping here. DTO classes in this project are used for representing HTTP request bodies.